### PR TITLE
Reeanbled Darwin

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: .github/workflows/build-macos.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      # Step: Install Python dependencies
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel
```
Comment (lucacarniato) [inline .github/workflows/build-macos.yml:65]: python -m pip install wheel numpy matplotlib pytest. you can use one line

File: .github/workflows/build-macos.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          python -m pip install numpy
+          python -m pip install matplotlib
+          python -m pip install pytest
+
+      # Step: Install system-provided dependencies
+      # macOS
+      - if: runner.os == 'macOS'
+        name: Install backend dependencies
+        run: | # brew update
+          brew install boost
```
Comment (lucacarniato) [inline .github/workflows/build-macos.yml:75]: brew install boost doxygen

File: .github/workflows/build-macos.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            # last possibility is being on a feature branch, which starts with "feature/"
+            # if not, it is not possile to continue
+            exit 1
+          fi
+          echo "BACK_END_BRANCH=$str" >> $GITHUB_ENV
+
+      # Step: Build Wheel
+      - name: Build Wheel
+        env:
+          CC: gcc-11
```
Comment (lucacarniato) [inline .github/workflows/build-macos.yml:102]: arent CC and CXX already defined above?

**Review by lucacarniato — APPROVED** [review]

File: .github/workflows/build-macos.yml (hunk 3 above)
Comment (ahmad-el-sayed) [reply .github/workflows/build-macos.yml:102]: Yes, and they should be removed up there.

File: .github/workflows/build-macos.yml (hunk 1 above)
Comment (ahmad-el-sayed) [reply .github/workflows/build-macos.yml:65]: Of course. It is nicer to have separate commands if errors happen.

**Pull request merged**